elements/userlevel/queuedevice.cc

```diff
@@ -102,8 +102,12 @@ int RXQueueDevice::parse(Vector<String> &conf, ErrorHandler *errh) {
     QueueDevice::parse(conf, errh);
 
 	_promisc = true;
+	_vlan_filter = true;
```

Comment:
Sorry I missed it up to now, but I would expect the default behavior to strip VLAN but not to filter VLANs. What would be filter filtering on by default? Can we set an index somehow?

Reply:
You're right, my bad.
What does "set an index" mean?

Reply:
Well the filtering feature allows to filter some VLAN ID I guess, right? 
I see the function rte_eth_dev_vlan_filter takes an ID. is The VLAN_FILTER parameter you added even useful without a way to set the ID being filtered?

Reply:
Yeah, I see what you mean... Well, I used rte_eth_dev_set_vlan_offload to set it in general (because my click configuration needs to get the entire range < 4096, so VLAN_FILTER is useful for me), but I think you're right and it would be wiser to fix it so it can be configured by VLAN id (and I'm guessing it's relevant for stripping as well)...I'll see how can I modify and test it.
Thanks a lot for your feedback!

Reply:
What is the default behavior when no ID is specifically set? If it accepts all VLAN, we can merge this as it and then you can extend. But I don't see where this is defined...

Reply:
Yes, it accepts all VLANs. I don't know if it counts as documentation but rte_eth_dev_set_vlan_offload [description](https://doc.dpdk.org/api/rte__ethdev_8h.html#a81a17c2c85724d5222ab137188599a28) is the best I could find... But maybe it's best to rewrite it with the support of filtering/stripping specific vlan ids.